pkg/server/server.go

```diff
@@ -166,6 +170,7 @@ func (s *Server) Start(errCh chan error) error {
 
 		for range interruptChannel {
 			log.Info("Shutting down")
+			s.BuildPoller.Stop()
```

Comment:
The poller should be stopped in the same function that started it. This would also remove the build poller dependency I believe